### DISCUSSION TITLE
Nilan EC9 with CTS 602

### DISCUSSION THREAD

**Discussion by sainio** [veista/nilan#54] [Q&A] [answered]
Hey I have this Nilan EC9 with CTS 602 and got your setup working now nicely but few question still pops to existence:

I can set the Nilan: Top Water Heater (Electric) from the GUI but could not do any automations to it, is it just me or is it missing from the code.  I used Kimmo Hyvönen's original python examples and with that I can set it manually by running that python script.

Other thing I was looking for is to set the Bottom water temp to control the compressor and room intake to control cooling/ventilation/heating as CTS 602 seems not to have away to control them directly.  All this I can do with my own script.

Any helps is greatly appreciated.

**Comment by veista** [comment, marked as answer]
What is your devices' modbus version? 

You can do automations from the automation menu and can try the device calls from the dev tools.

If you had button emulation in your config it is not possible with this integration without getting an update from Nilan.

**Reply by sainio** [reply]
This is Modbus version 21, I have some software update too but I have not installed it.

**Reply by veista** [reply]
Ok. With modbus version 21 pretty much everything should work.

**Reply by veista** [reply]
https://www.home-assistant.io/docs/automation/basics/

**Reply by sainio** [reply]
Okay, but if I choose Device: Nilan from the long list of options I cannot find Top Water Heater (Electric) setting to change the temperature.  If I use the Turn on Option, it does nothing if the set temperature is lower than the request.

From the Overview window settings can be changed, except then fan speed.

![image](https://user-images.githubusercontent.com/8513998/224226799-ae1caf1b-d91a-4d28-bd08-f499514e5173.png)

**Reply by veista** [reply]
Fan speed is under climate entity.

**Reply by veista** [reply]
![Screenshot_20230310_115452_Samsung Internet](https://user-images.githubusercontent.com/91916713/224285547-eb8089fb-b99d-4f8d-93a2-e09cdec94615.jpg)

Everything works as it should. I think you need to get more familiar with Home Assistant.

**Reply by sainio** [reply]
I'm trying :-)  Did not realize a normal automation would make me a developer, as settings has this automations tab too.  Found it now, thanks, I'll leave you to your duties now. Thanks.

**Comment by veista** [comment]
Btw. Can you please help with #9
